### PR TITLE
Optimise metadata and offsets cache

### LINKED ISSUE #32: Fetch group offsets from kafka directly
Add new endpoint `/api/v2/counsumer/<group id>` to fetch group offsets from kafka directly via `brod_utils:fetch_committed_offsets/4`.

### LINKED ISSUE #35: Let the application run when zookeeper is unavailable
We only need zookeeper to show topic config.
It should be possible for KastleX to start successfully when zookeeper endpoints are not configured, or not available.

### PR DESCRIPTION
**Pull request by id**
- get rid of zookeeper dependency
- fetch and cache metadata and offsets more efficiently
- new /api/v2/consumer endpoint which fetches info about consumers directly from Kafka

Fixes #32 
Fixes #35

### CODE REVIEW

**Comment by zmstone** [comment]
Sorry for the late review. We can release this in 1.6.
Perhaps rebase on brod 3.7 later.

**Comment by id** [comment]
ping @zmstone

File: lib/kastlex/cg_cache.ex [hunk 1]
```diff
@@ -133,19 +133,9 @@ defmodule Kastlex.CgCache do
   defp put(x, key, val) when is_map(x), do: Map.put(x, key, val)
   defp put(x, key, val) when is_list(x), do: Keyword.put(x, key, val)
 
-  # this is kept for backward compatibility
```
Comment (zmstone) [inline lib/kastlex/cg_cache.ex]: why is this deleted ?

Comment (zmstone) [reply lib/kastlex/cg_cache.ex]: nvm, moved.

File: lib/kastlex/offsets_cache.ex [hunk 2]
```diff
@@ -76,28 +95,29 @@ defmodule Kastlex.OffsetsCache do
...
-      Enum.each(responses,
-                fn(tr) ->
-                  Enum.each(tr[:partition_responses],
-                            fn(pr) ->
-                              try do
-                                [offset] = pr[:offsets]
-                                :ets.insert(@table, {{tr[:topic], pr[:partition]}, offset})
+    leader = Kastlex.MetadataCache.get_leader(id)
+    client_id = Kastlex.get_brod_client_id()
+    {:ok, conn} = :brod_client.get_connection(client_id, String.to_charlist(leader.host), leader.port)
```
Comment (zmstone) [inline lib/kastlex/offsets_cache.ex]: this is pretty easy to crash if a node is taken down but the metadata is not refreshed yet.
perhaps a better approach is to spawn a worker for each leader and make hem run independently with a `timer:send_interval` (and ignore `{:error, reason}` too).

File: lib/kastlex/offsets_cache.ex [hunk 3]
```diff
@@ -76,28 +95,29 @@ defmodule Kastlex.OffsetsCache do
                          end)
     request_fields = [replica_id: -1, topics: topics]
     request = :kpro.make_request(:list_offsets, 0, request_fields)
```
Comment (zmstone) [inline lib/kastlex/offsets_cache.ex]: pick version ?

File: lib/kastlex/offsets_cache.ex [hunk 4]
```diff
@@ -76,28 +95,29 @@ defmodule Kastlex.OffsetsCache do
...
+    %{state | :kafka_requests => new_map}
+  end
+
+  defp handle_offsets_response(msg) do
+    responses = msg[:responses]
+    Enum.each(responses,
+              fn(tr) ->
+                Enum.each(tr[:partition_responses],
+                          fn(pr) ->
+                            try do
```
Comment (zmstone) [inline lib/kastlex/offsets_cache.ex]: This is from old code, but perhaps refine it now:
this `try do` is too brutal, a better approach is to respect the `pr.error_code`.
if `:no_error`, bind `offset` with `[offset] = pr[:offsets]` or `offset = pr.offset` depending on the request version.

File: lib/kastlex/offsets_cache.ex [hunk 5]
```diff
@@ -24,80 +26,97 @@ defmodule Kastlex.OffsetsCache do
...
+          end)
+        client_id = Kastlex.get_brod_client_id()
+        leader = Kastlex.MetadataCache.get_leader(id)
+        {:ok, conn} = :brod_client.get_connection(client_id, String.to_charlist(leader.host), leader.port)
+        vsn = :brod_kafka_apis.pick_version(conn, :list_offsets)
+        request_fields = [replica_id: -1, topics: topics]
+        request = :kpro.make_request(:list_offsets, vsn, request_fields)
+        {:ok, kpro_rsp(msg: msg)} = :kpro.request_sync(conn, request, 10_000)
+        handle_offsets_response(msg[:responses])
+        :erlang.send_after(refresh_timeout_ms, parent, {@refresh, id, generation})
```
Comment (zmstone) [inline lib/kastlex/offsets_cache.ex]: we need a big `try catch` for the spawned `fn`, and only do this `send_after` in `after` clause.

Comment (id) [reply lib/kastlex/offsets_cache.ex]: @zmstone fixed
